web/src/components/CountryDistribution/CountryDistributionPage.tsx

```diff
@@ -110,9 +108,30 @@ export function CountryDistributionPage() {
   }, [setPlaces])
 
   const IntroContent = useMemo(() => {
-    const contentFilename = getPerCountryIntroContentFilename(currentRegion)
+    const contentFilename = getPerCountryIntroContentFilename(region)
     return getRegionPerCountryContent(contentFilename)
-  }, [currentRegion])
+  }, [region])
+
+  const setRegion = useCallback(
```

Comment:
instead of directly mutating the React `region` state, this will instead just mutate the query params. Once it is updated, the React state will recompute accordingly